### PR TITLE
codefresh: init at 0.87.3

### PR DESCRIPTION
**Pull request by takac**
## Description of changes

Add codefresh cli[1] tool to interact with codefresh[2] service.

[1] https://github.com/codefresh-io/cli
[2] https://codefresh.io/

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by takac** [comment]
I had an issue running the tests `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` which I couldn't solve, seemed to be an issue with generating the name. 
```
error:
       … while evaluating the 'name' attribute of a derivation

         at «none»:0: (source not available)

       … from call site

         at /Users/tom.cammann/.cache/nixpkgs-review/rev-76b34d5d1c23cdf798d29de00cf78b46c8ab50dc/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:371:9:

          370|         in
          371|         lib.strings.sanitizeDerivationName (
             |         ^
          372|           if attrs ? name

       … while calling anonymous lambda

         at /Users/tom.cammann/.cache/nixpkgs-review/rev-76b34d5d1c23cdf798d29de00cf78b46c8ab50dc/nixpkgs/lib/strings.nix:1174:3:

         1173|   in
         1174|   string:
             |   ^
         1175|   # First detect the common case of already valid strings, to speed those up

       … from call site

         at /Users/tom.cammann/.cache/nixpkgs-review/rev-76b34d5d1c23cdf798d29de00cf78b46c8ab50dc/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:273:18:

          272|       version = attrs.version or package.version;
          273|       baseName = unlessNull name "${safeName}-${version}";
             |                  ^
```
Any help appreciated.

**Comment by AndersonTorres** [comment]
Redo everything according to the contributing guidelines.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -17735,6 +17735,12 @@
     githubId = 321799;
     name = "Paul Colomiets";
   };
+  takac = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:17738]: Split.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -519,6 +519,8 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Security CoreServices;
   };
 
+  codefresh = callPackage ../by-name/co/codefresh/package.nix { };
+
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:523]:
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (takac) [reply maintainers/maintainer-list.nix:17738]: Split what exactly?

Comment (takac) [reply maintainers/maintainer-list.nix:17738]: https://github.com/NixOS/nixpkgs/pull/290907

Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:17738]: No need. Just create a commit.

**Comment by AndersonTorres** [comment]
I can't understand the error of ofBorg...

**Comment by AndersonTorres** [comment]
Usually I "solve" this by rebasing over the most recent upstream/master.

File: pkgs/by-name/co/codefresh/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  doDist = false;
+
+  passthru.tests.version = testers.testVersion {
+    package = codefresh;
+    # codefresh needs to read a config file, this is faked out with a subshell
+    command = "codefresh --cfconfig <(echo 'contexts:') version";
+  };
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/by-name/co/codefresh/package.nix:27]: Tips:

1. remove `with lib;`
2. Keep meta attributes sorted alphabetically

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/co/codefresh/package.nix (hunk 3 above)
Comment (takac) [reply pkgs/by-name/co/codefresh/package.nix]: Doesn't seem to have helped with the failing OfBorg test? Any ideas on how to fix that

**Comment by takac** [comment]
Adding `package.json` and `packageJSON = ./package.json;` was the solution, despite this building fine locally and the default for `mkYarnPackage` having a sensible default.

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Thank you.

**Pull request merged**